### PR TITLE
OI-71: Implemented a slideshow, and another slideshow on image click.

### PR DESCRIPTION
**Pull request by nvelychenko**

### CODE REVIEW

File: modules/openideal_slideshow/js/slider-configuration.js [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+  /**
+   * Init the photoswipe slider.
+   *
+   * @param {string} gallerySelector
+   *   The index of the current element.
+   * @param {object} mySwiper
+   *   The slick HTML element.
+   */
+  var initPhotoSwipeFromDOM = function (gallerySelector, mySwiper) {
```
Comment (db-Matroskeen) [inline modules/openideal_slideshow/js/slider-configuration.js:48]: 1) Please add a link to the original example: https://photoswipe.com/documentation/getting-started.html
2) Please move custom Drupal behavior to the top of the file.

File: modules/openideal_slideshow/js/slider-configuration.js [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+   * Slideshow behaviour.
+   *
+   * @type {Drupal~behavior}
+   *
+   * @prop {Drupal~behaviorAttach} attach
+   *   Attach slick carousel to Slideshow block.
+   */
+  Drupal.behaviors.OpenidealSlideshowSlickConfig = {
+    attach: function (context, settings) {
+      $('.openideal-slideshow').once('kekw').each(function () {
```
Comment (db-Matroskeen) [inline modules/openideal_slideshow/js/slider-configuration.js]: 1) Can we use `context`?
2) kekw?

File: modules/openideal_slideshow/js/slider-configuration.js [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+
+  /**
+   * Slideshow behaviour.
+   *
+   * @type {Drupal~behavior}
+   *
+   * @prop {Drupal~behaviorAttach} attach
+   *   Attach slick carousel to Slideshow block.
+   */
+  Drupal.behaviors.OpenidealSlideshowSlickConfig = {
```
Comment (db-Matroskeen) [inline modules/openideal_slideshow/js/slider-configuration.js]: Probably this is not Slick anymore.

File: modules/openideal_slideshow/js/slider-configuration.js [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+    }
+  };
+
+  /**
+   * Slideshow behaviour.
+   *
+   * @type {Drupal~behavior}
+   *
+   * @prop {Drupal~behaviorAttach} attach
+   *   Attach slick carousel to Slideshow block.
```
Comment (db-Matroskeen) [inline modules/openideal_slideshow/js/slider-configuration.js]: Probably this is not Slick anymore.

File: modules/openideal_slideshow/templates/slideshow.html.twig [hunk 5]
```diff
@@ -1,5 +1,77 @@
+{# The swiper slideshow. #}
```
Comment (db-Matroskeen) [inline modules/openideal_slideshow/templates/slideshow.html.twig]: 1) I think we can remove the following line from `openideal_slideshow_theme()`:
`'template' => 'slideshow',`

In that case, template will have default name - `openideal-slideshow.html.twig`.

2) I don't see how do we handle cache metadata of the slideshow images. Please make sure the build array has appropriate cache tags/contexts.

File: modules/openideal_slideshow/templates/slideshow.html.twig [hunk 6]
```diff
@@ -1,5 +1,77 @@
...
 <div class="openideal-slideshow">
-    {% for item in items %}
-        <img src="{{ item.uri.value | image_style('slideshow_large') }}" alt="{{ item._referringItem.alt }}">
-    {% endfor %}
+    <ul class="swiper-wrapper my-gallery" itemscope itemtype="http://schema.org/ImageGallery">
+        {% for key, item in items %}
+            <li id="{{ key + 1 }}" class="swiper-slide" itemprop="associatedMedia" itemscope
```
Comment (db-Matroskeen) [inline modules/openideal_slideshow/templates/slideshow.html.twig]: What IDs will we have in this case? 1, 2, 3?
Can we add some labels to make them more meaningful? (i.e. `slide-N`)?

File: modules/openideal_slideshow/templates/slideshow.html.twig [hunk 7]
```diff
@@ -1,5 +1,77 @@
...
+                itemtype="http://schema.org/ImageObject">
+                {# For the photoswipe use default image without styles. #}
+                <a id="{{ key == 0 ? 'first' : '' }}" title="click to zoom-in" href="{{ file_url(item.uri.value) }}"
+                   itemprop="contentUrl" data-size="1200x600">
+                    <img itemprop="thumbnail" src="{{ item.uri.value | image_style('slideshow_large') }}"
```
Comment (db-Matroskeen) [inline modules/openideal_slideshow/templates/slideshow.html.twig]: Should we add item attributes to the markup?
See default core `ImageFormatter.php` for reference.

File: modules/openideal_slideshow/templates/slideshow.html.twig [hunk 8]
```diff
@@ -1,5 +1,77 @@
...
 <div class="openideal-slideshow">
-    {% for item in items %}
-        <img src="{{ item.uri.value | image_style('slideshow_large') }}" alt="{{ item._referringItem.alt }}">
-    {% endfor %}
+    <ul class="swiper-wrapper my-gallery" itemscope itemtype="http://schema.org/ImageGallery">
+        {% for key, item in items %}
+            <li id="{{ key + 1 }}" class="swiper-slide" itemprop="associatedMedia" itemscope
+                itemtype="http://schema.org/ImageObject">
+                {# For the photoswipe use default image without styles. #}
+                <a id="{{ key == 0 ? 'first' : '' }}" title="click to zoom-in" href="{{ file_url(item.uri.value) }}"
```
Comment (db-Matroskeen) [inline modules/openideal_slideshow/templates/slideshow.html.twig]: Labels/titles and other strings should be wrapped with t() function or similar.

File: modules/openideal_slideshow/templates/slideshow.html.twig [hunk 9]
```diff
@@ -1,5 +1,77 @@
...
+                   itemprop="contentUrl" data-size="1200x600">
```
Comment (db-Matroskeen) [inline modules/openideal_slideshow/templates/slideshow.html.twig]: Why 1200x600? (just curious)

File: modules/openideal_slideshow/templates/slideshow.html.twig [hunk 10]
```diff
@@ -1,5 +1,77 @@
...
+        {% endfor %}
+    </ul>
+</div>
+
+
+{#
+  Add PhotoSwipe (.pswp) element to DOM -
+  Root element of PhotoSwipe. Must have class pswp.
+#}
+<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">
```
Comment (db-Matroskeen) [inline modules/openideal_slideshow/templates/slideshow.html.twig]: 1) Can we move this markup to a separate template and include it here?
2) t() function for labels

**Review by db-Matroskeen — CHANGES_REQUESTED** [review]

File: modules/openideal_slideshow/templates/slideshow.html.twig (hunk 9 above)
Comment (nvelychenko) [reply modules/openideal_slideshow/templates/slideshow.html.twig]: Forgot to add the appropriate sizes from the image entity. Fixed.

File: modules/openideal_slideshow/templates/slideshow.html.twig (hunk 10 above)
Comment (nvelychenko) [reply modules/openideal_slideshow/templates/slideshow.html.twig]: 1. Done

File: modules/openideal_slideshow/js/slider-configuration.js (hunk 2 above)
Comment (nvelychenko) [reply modules/openideal_slideshow/js/slider-configuration.js]: 1. Done.
2. ..

File: modules/openideal_slideshow/templates/slideshow.html.twig (hunk 6 above)
Comment (nvelychenko) [reply modules/openideal_slideshow/templates/slideshow.html.twig]: Done.

**Pull request merged**